### DISCUSSION TITLE
Best way to define units for DynamicQuantities in a Pluto notebook

### DISCUSSION THREAD

**Discussion by alhirzel** [fonsp/Pluto.jl#3105] [Questions about using Pluto]
The package [DynamicQuantities.jl](https://github.com/SymbolicML/DynamicQuantities.jl) allows a user to [define custom units](https://github.com/SymbolicML/DynamicQuantities.jl?tab=readme-ov-file#custom-units):

```julia
@register_unit lbf 4.4482u"N"   # allows you to write things like 10u"lbf"
```

It is sensible that this "custom" namespace of units isn't monitored by Pluto: how could Pluto possibly out-craft all the crafty fox macro writers of the world? Having understood this, what is the best way to make sure the definition above occurs before `u"lbf"` is used? Is there any way to tag the cell containing the invocation of `@register_unit` as the defining cell for `u"lbf"` or something like that? I read briefly about some macro analysis that Pluto performs; is there some kind of heuristic cat-and-mouse trick I can PR an improvement against?

(Note that this question is similar to https://github.com/fonsp/Pluto.jl/discussions/1374, but `DynamicQuantities.jl` does not require the same `module` shenanigans mentioned in that discussion.)

**Comment by cstjean** [comment]
If you're using `lbf` instead of `u"lbf"`, doesn't it work?
